### DISCUSSION TITLE
OpenCascade wasm/js library size

### DISCUSSION THREAD

**Discussion by stewartoallen** [zalo/CascadeStudio#67] [General]
The size of the WASM and JS to support this package is considerably smaller that the two files in the dist directory of the npm package. How did you produce the smaller files? By omitting some of the OCC apis?

An old hat at JS, but new to OCC and WASM, so please bear with me.

Thanks!

**Comment by Irev-Dev** [comment]
The module listed in package.json is a little misleading. It a different fork of opencascade.js, maybe there's more to it, but I think that's probably the explanation @stewartoallen.
https://github.com/zalo/opencascade.js

**Reply by stewartoallen** [reply]
I see now. Thanks. It uses a modified`opencascade.idl` to limit the APIs exposed. And that makes the size and load time manageable. The parent project exposes everything which is totally unworkable for a web app.

**Reply by Irev-Dev** [reply]
There's a bunch of relevant context in this PR too @stewartoallen 
https://github.com/zalo/CascadeStudio/pull/12

**Reply by stewartoallen** [reply]
thanks for that pointer. the increased wasm size is definitely a problem and not really because of network transport. it takes about 40 seconds to instantiate the larger wasm *after* the bytes are loaded. the version being used in CascadeStudio takes about 7 seconds.
